testsuite/tests/kuadrant/gateway/test_authpolicy_to_gateway.py

```diff
@@ -0,0 +1,60 @@
+"""Test for AuthPolicy attached directly to gateway"""
+from time import sleep
+import pytest
+
+from testsuite.openshift.objects.auth_config.auth_policy import AuthPolicy
+from testsuite.openshift.objects.gateway_api.route import HTTPRoute
+
+
+@pytest.fixture(scope="module")
+def gateway_httproute(request, gateway, wildcard_domain, module_label, blame):
+    """HTTPRoute with wildcard domain"""
+    route = HTTPRoute.create_instance(gateway.openshift, blame("gw-route"), gateway, {"app": module_label})
+    route.add_hostname(wildcard_domain)
+    route.set_path_match("/")
+
+    request.addfinalizer(route.delete)
+    route.commit()
+    return route
+
+
+@pytest.fixture(scope="module")
+def gateway_authorization(request, gateway, authorization_name, openshift, module_label):
+    """AuthPolicy attached straight to gateway"""
+    auth_policy = AuthPolicy.create_instance(
+        openshift, f"gw-{authorization_name}", gateway, labels={"testRun": module_label}
+    )
+    auth_policy.authorization.add_opa_policy("deny-all", "allow { false }")
+    request.addfinalizer(auth_policy.delete)
+    return auth_policy
+
+
+@pytest.fixture(scope="module")
+def client_to_deny(gateway, exposer, blame):
+    """Hostname that should be denied by AuthPolicy attached directly to gateway"""
+    hostname = exposer.expose_hostname(blame("not-accepted-hostname"), gateway)
+    client = hostname.client()
+    yield client
+    client.close()
+
+
+def test_authpolicy_attached_to_gateway(gateway_httproute, gateway_authorization, client, client_to_deny):
+    # pylint: disable=unused-argument
+    """
+    Test if AuthPolicy denying all requests sent to the undefined domains:
+        - send request to the hostname defined
+        - send request to the undefined domain without AuthPolicy attached to gateway
+        - commit deny-all AuthPolicy attached to gateway
+        - send request to the undefined domain with deny-all AuthPolicy attached to gateway
```

Comment:
I am not sold on this being a good test case, what exactly are you trying to test? That AuthPolicy is correctly applied for Gateway or is there something else I am not seeing?